vmdb/app/models/ems_openshift.rb

```diff
@@ -0,0 +1,34 @@
+class EmsOpenshift < EmsContainer
+  include ContainerProviderMixin
+
+  # TODO: what about k8s entities?
+  has_many :container_routes,                      :foreign_key => :ems_id, :dependent => :destroy
+  has_many :container_projects,                    :foreign_key => :ems_id, :dependent => :destroy
+
+  default_value_for :port, 8443
+
+  def self.ems_type
+    @ems_type ||= "openshift".freeze
+  end
+
+  def self.description
+    @description ||= "Openshift".freeze
```

Comment:
OpenShift is a capital O and capital S.  I'm ok with the constants like `EmsOpenshift` and `OpenshiftClient` since the underscore/lowercase (i.e. openshift) matches with the upstream naming, and users generaly won't "see" those words.  However any place that's UI facing should say "OpenShift".  To take it a step further, I think the official naming must be one of "Red Hat OpenShift" or "OpenShift by Red Hat" (you'd probably have to ask what the right string is)